sycl/test-e2e/Basic/multi_ptr_null_relational_operators.cpp

```diff
@@ -0,0 +1,55 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/sycl.hpp>
+
+template <typename multi_ptr_t> void nullptrRelationalOperatorTest() {
+  constexpr int OUTPUT_SIZE = 8;
+  bool output[OUTPUT_SIZE];
+  try {
+    sycl::queue queue;
+    sycl::buffer<bool, 1> buf(output, sycl::range<1>(OUTPUT_SIZE));
+    queue
+        .submit([&](sycl::handler &cgh) {
+          auto dev_acc = buf.get_access<sycl::access::mode::write>(cgh);
+          sycl::local_accessor<bool, 1> locAcc(1, cgh);
+          cgh.parallel_for(sycl::nd_range<1>{1, 1}, [=](sycl::id<1>) {
+            locAcc[0] = 10;
+            multi_ptr_t mp(locAcc);
+            dev_acc[0] = std::less<multi_ptr_t>()(nullptr, mp) == nullptr < mp;
```

Comment:
Should we also check the result using plain decorated pointers without `multi_ptr`?

Reply:
I'm not quite sure I understand what you mean by "decorated pointers without multiptr." Could you please provide more context?

Reply:
`decltype(multi_ptr_obj.get_decorated())` I would be something like `int __attribute__((opencl)) *`.

Reply:
@aelovikov-intel Testing plain decorated pointers is beyond the scope of this PR, as it exclusively addresses relational operators within multi_ptrs.